### ISSUE TITLE
PolyPen does not work in Local View and wrecks BGL drawing

### ISSUE THREAD

**Issue by carter2422** [CGCookie/retopoflow#251]
Activating Polypen in Local View on the source object wrecks the whole Blender UI and gives this error:
```
EXCEPTION (<class 'AttributeError'>): 'CGC_Polypen' object has no attribute 'tar_bmeshrender'
         /Users/jonathanwilliamson/blender-dev/blender-master/build_versions/v2.77a/bin/blender.app/Contents/MacOS/./../Resources/2.77/scripts/modules/bpy_types.py
000 0597:__getattribute__() return super().__getattribute__(attr)
         /Users/jonathanwilliamson/Dropbox (Personal)/blenderDev/addons/cgCookie/scripts/addons/retopoFlow/op_polypen/polypen_modal.py
001 0230:draw_postview() self.tar_bmeshrender.draw(opts=self.render_normal)
         /Users/jonathanwilliamson/Dropbox (Personal)/blenderDev/addons/cgCookie/scripts/addons/retopoFlow/modaloperator.py
002 0161:draw_callback_postview() self.draw_postview(context)
```

Contours and Polystrips both work fine.

**Comment by vxlcoder** [comment]
polypen will create a new target object at start if one does not exist.  then, it tries to enter edit mode.  but, if the user is in local view on another object, we cannot edit an object that isn't in local view.  according to http://blenderartists.org/forum/showthread.php?382191-managing-local-view-from-python, it seems that there is not a way to add an object to local view.  so, the "quick and dirty fix" is to try entering edit mode, but if a TypeError is thrown, jump in and out of local view and try edit mode again.  need to check with blender devs to see if there is a better way to handle local view.

**Comment by vxlcoder** [comment]
@carter2422 this issue _should_ be fixed!  please test when you get a chance.

**Comment by carter2422** [comment]
This appears to be working really well @vxlcoder. Nice work. Merged into 1.2.2
